Loop/Extensions/DeviceDataManager+DeviceStatus.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  DeviceDataManager+DeviceStatus.swift
+//  Loop
+//
+//  Created by Nathaniel Hamming on 2020-07-10.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import LoopKit
+import LoopKitUI
+
+extension DeviceDataManager {
+    var cgmStatusHighlight: DeviceStatusHighlight? {
+        if bluetoothState == .poweredOff {
+            return BluetoothStateManager.bluetoothOffHighlight
+        } else if bluetoothState == .denied ||
+            bluetoothState == .unauthorized
+        {
+            return BluetoothStateManager.bluetoothUnavailableHighlight
+        } else if cgmManager == nil {
+            return DeviceDataManager.addCGMStatusHighlight
+        } else {
+            return (cgmManager as? CGMManagerUI)?.cgmStatusHighlight
+        }
+    }
+    
+    var cgmLifecycleProgress: DeviceLifecycleProgress? {
+        return (cgmManager as? CGMManagerUI)?.cgmLifecycleProgress
+    }
+    
+    var pumpStatusHighlight: DeviceStatusHighlight? {
+        if bluetoothState == .denied ||
+            bluetoothState == .unauthorized ||
+            bluetoothState == .poweredOff
+        {
+            return BluetoothStateManager.bluetoothEnableHighlight
+        } else if pumpManager == nil {
+            return DeviceDataManager.addPumpStatusHighlight
+        } else {
+            return pumpManagerStatus?.pumpStatusHighlight
+        }
+    }
+    
+    var pumpLifecycleProgress: DeviceLifecycleProgress? {
+        return pumpManagerStatus?.pumpLifecycleProgress
+    }
+    
+    static var addCGMStatusHighlight: AddDeviceStatusHighlight {
+        return AddDeviceStatusHighlight(localizedMessage: NSLocalizedString("Add CGM", comment: "Title text for button to set up a CGM"))
+    }
+    
+    static var addPumpStatusHighlight: AddDeviceStatusHighlight {
+        return AddDeviceStatusHighlight(localizedMessage: NSLocalizedString("Add Pump", comment: "Title text for button to set up a Pump"))
+    }
+    
+    struct AddDeviceStatusHighlight: DeviceStatusHighlight {
+        var localizedMessage: String
+        var imageSystemName: String = "plus.circle"
+        var state: DeviceStatusHighlightState = .normal
+    }
+    
+    func didTapOnPumpStatus(_ view: BaseHUDView? = nil) -> HUDTapAction? {
```

Comment:
[nit] Really minor nit, but consider ordering of CGM and Pump functionality in the same order through out file. Above you have CGM before Pump, but here you have Pump then CGM.